okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/http/RealInterceptorChain.kt

```diff
@@ -98,6 +108,22 @@ class RealInterceptorChain(
     return copy(writeTimeoutMillis = checkDuration("writeTimeout", timeout.toLong(), unit))
   }
 
+  override fun withDns(dns: Dns?): Interceptor.Chain =
+    withClientOverrides {
+      copy(dns = dns)
+    }
+
+  override fun withSocketFactory(socketFactory: SocketFactory?): Interceptor.Chain =
+    withClientOverrides {
+      copy(socketFactory = socketFactory)
+    }
+
+  private fun withClientOverrides(overrides: ClientOverrides.() -> ClientOverrides): Interceptor.Chain {
+    check(exchange == null) { "ClientOverrides can't be adjusted in a network interceptor" }
```

Comment:
nit: the caller doesn’t know what a ClientOverrides is . . . maybe pass a string like "socketFactory" into this function, so we can throw an error message the caller will understand?